app/models/concerns/form526_rapid_ready_for_decision_concern.rb

```diff
@@ -94,27 +78,44 @@ def diagnostic_codes
   end
 
   def prepare_for_evss!
-    return unless forward_to_mas?
+    return if pending_eps? || disability_not_service_connected?
 
-    save_metadata(forward_to_mas: true)
-    insert_classification_codes
+    save_metadata(forward_to_mas_all_claims: true) if Flipper.enabled?(:rrd_mas_all_claims_tracking) &&
+                                                      !single_issue_hypertension_cfi?
+
+    if Flipper.enabled?(:rrd_mas_disability_tracking) && single_disability_eligible_for_mas?
+      save_metadata(forward_to_mas: true)
+      insert_classification_codes
+    end
   end
 
   def send_post_evss_notifications!
     send_completed_notification if rrd_job_selector.rrd_applicable?
-    notify_mas if form.dig('rrd_metadata', 'forward_to_mas')
+    notify_mas
```

Comment:
I suggest renaming this to `conditionally_notify_mas`

Reply:
Great suggestion. In the interest of expediency, I'll leave this for the next PR (which will likely land this week).